File: NetFramework/TreeView/VB/CustomEditors/CustomColorPicker.vb
Imports C1.Win.TreeView

Public Class CustomColorPicker
    Inherits C1.Win.C1Input.C1ColorPicker
    Implements IC1TreeEditor

    Public Overrides Sub C1EditorInitialize(value As Object, attrs As IDictionary) Implements IC1TreeEditor.C1EditorInitialize
        AutoSize = False
        TabStop = False

        If (attrs.Contains("BackColor")) Then
            BackColor = CType(attrs("BackColor"), Color)
        End If
        If (attrs.Contains("Font")) Then
            Font = CType(attrs("Font"), Font)
        End If
        If (attrs.Contains("ForeColor")) Then
            ForeColor = CType(attrs("ForeColor"), Color)
        End If
        If (attrs.Contains("DisabledForeColor")) Then
            ForeColor = CType(attrs("DisabledForeColor"), Color)
        End If
        If (attrs.Contains("MaxLength")) Then
            MaxLength = CType(attrs("MaxLength"), Integer)
        End If
        Color = value
    End Sub

    Public Sub C1EditorSetFormat(format As String) Implements IC1TreeEditor.C1EditorSetFormat

    End Sub

    Public Overrides Sub C1EditorUpdateBounds(rc As Rectangle) Implements IC1TreeEditor.C1EditorUpdateBounds
        Bounds = rc
    End Sub

    Public Overrides Function C1EditorGetValue() As Object Implements IC1TreeEditor.C1EditorGetValue
        Return Color
    End Function

    Public Overrides Function C1EditorKeyDownFinishEdit(e As KeyEventArgs) As Boolean Implements IC1TreeEditor.C1EditorKeyDownFinishEdit
        If (e.KeyData = Keys.Enter) Then
            Return True
        End If
        Return False
    End Function

    Public Overrides Function C1EditorValueIsValid() As Boolean Implements IC1TreeEditor.C1EditorValueIsValid
        Return True
    End Function
End Class

File: NetFramework/TreeView/CS/CustomEditors/CustomColorPicker.cs
using C1.Win.C1Input;
using C1.Win.TreeView;
using System.Collections;
using System.Drawing;
using System.Windows.Forms;
using System;

namespace CustomEditors
{
    public class CustomColorPicker: C1ColorPicker, IC1TreeEditor
    {
        public override object C1EditorGetValue()
        {
            return Color;
        }

        public override void C1EditorInitialize(object value, IDictionary attrs)
        {            
            AutoSize = false;
            TabStop = false;
            
            if (attrs.Contains("BackColor"))
                BackColor = (Color)attrs["BackColor"];
            if (attrs.Contains("Font"))
                Font = (Font)attrs["Font"];
            if (attrs.Contains("ForeColor"))
                ForeColor = (Color)attrs["ForeColor"];
            if (attrs.Contains("DisabledForeColor"))
                ForeColor = (Color)attrs["DisabledForeColor"];
            if (attrs.Contains("MaxLength"))
                MaxLength = (int)attrs["MaxLength"];
        
            Color = (Color)value;
        }

        public override bool C1EditorKeyDownFinishEdit(KeyEventArgs e)
        {
            if (e.KeyData == Keys.Enter)
                return true;
            return false;
        }

        public void C1EditorSetFormat(string format)
        {

        }

        public override void C1EditorUpdateBounds(Rectangle rc)
        {
            Bounds = rc;
        }

        public override bool C1EditorValueIsValid()
        {
            return true;
        }
    }
}
